docs/topics/patterns/instant-messaging.mdx

```diff
@@ -0,0 +1,359 @@
+    topicClient
+      .subscribe(
+        TopicExample.CACHE_NAME,
+        TOPIC_NAME,
+        new ISubscriptionCallbacks() {
+          @Override
+          public void onItem(TopicMessage message) {
+            logger.info("Received message on topic {}: {}", TOPIC_NAME, message.toString());
+          };
+
+          @Override
+          public void onError(Throwable error) {
+            logger.error("Subscription to topic {} failed with error", TOPIC_NAME, error);
+          };
+        };).join();
+  ```
+  </TabItem>
+  <TabItem value="dotnet" label=".NET">
+
+  ```csharp
+  var subscriptionTask = Task.Run(async () =>
+  {
+    var subscribeResponse = await topicClient.SubscribeAsync(cacheName, TopicName);
+    switch (subscribeResponse)
+    {
+      case TopicSubscribeResponse.Subscription subscription:
+        try {
+          var cancellableSubscription = subscription.WithCancellation(cts.Token);
+          await foreach (var message in cancellableSubscription) {
+            switch (message)
+            {
+              case TopicMessage.Binary:
+                Logger.LogInformation("Received binary message from topic: {binaryData}", Convert.ToBase64String(binary.Value));
+                break;
+              case TopicMessage.Text text:
+                Logger.LogInformation("Received string message from topic: {message}", text.Value);
+                break;
+              case TopicMessage.Error error:
+                Logger.LogInformation("Received error message from topic: {error}", error.Message);
+                cts.Cancel();
+                break;
+            }
+          }
+        } finally {
+          subscription.Dispose();
+        }
+
+        break;
+      case TopicSubscribeResponse.Error error:
+        Logger.LogInformation("Error subscribing to a topic: {error}", error.Message);
+        cts.Cancel();
+        break;
+    }
+  });
+  ```
+  </TabItem>
+</Tabs>
+
+## Security
+
+Momento provides a [robust authorization mechanism](../develop/authentication/tokens) that limits consumers to specific resources and actions. When building an instant messaging feature, it is incredibly important to take scope into consideration so users do not intercept messages they shouldn't have access to.
+
+The pattern described above uses both cache and topic resources. Following the best practice recommended above by making our chat room the topic name, we can create policies in a couple of different ways.
+
+### Client-side only
+
+It's possible to build instant messaging completely on the client side, meaning directly in the browser or mobile application. The policy below would be used in your clients.
+
+```json
+{
+  "permissions": [
+    {
+      "role": "readwrite",
+      "cache": "instant-messaging",
+      "item": {
+        "key": "{chatid}"
+      }
+    },
+    {
+      "role": "publishsubscribe",
+      "cache": "instant-messaging",
+      "topic": "{chatid}"
+    }
+  ]
+}
+```
```

Comment:
what is this? I don't think we actually support any JSON representation of policies, and this section of the doc doesn't really explain to a user how they would achieve this policy. Maybe that's okay if you are just trying to convey the theoretical concepts but I don't think a user could take this and get an app working from it, don't know if that is part of the goal here?